pkg/lib/collections/priority_queue_test.go

```diff
@@ -0,0 +1,89 @@
+//go:build unit || !integration
+
+package collections_test
+
+import (
+	"testing"
+
+	"github.com/bacalhau-project/bacalhau/pkg/lib/collections"
+	"github.com/stretchr/testify/suite"
+)
+
+type PriorityQueueSuite struct {
+	suite.Suite
+}
+
+func TestPriorityQueueSuite(t *testing.T) {
+	suite.Run(t, new(PriorityQueueSuite))
+}
+
+func (s *PriorityQueueSuite) TestSimple() {
+	type testcase struct {
+		v string
+		p int
+	}
+	testcases := []testcase{
+		{"A", 3},
+		{"A", 3},
+		{"B", 2},
+		{"C", 1},
+		{"C", 1},
+	}
+
+	pq := collections.NewPriorityQueue[string]()
+	for _, tc := range testcases {
+		pq.Enqueue(tc.v, tc.p)
+	}
```

Comment:
While testing I like to enqueue in an order that is different than the expected dequeue order to make sure we our queue is based on the priority and not on insertion order. So maybe:

```
inputs := []testcases{{"B", 2}, {"A", 3}, {"C", 1}, {"A", 3}, {"C", 1}, {"B", 2}}
expected := []testcases{{"A", 3},  {"A", 3}, {"B", 2}, {"B", 2} , {"C", 1},{"C", 1}}
```
```